### PR TITLE
Mutagen cleanup

### PR DESCRIPTION
**Pull request by lazka**

### CODE REVIEW

**Comment by jrobeson** [comment]
@lazka : it's failing flake8, can you look into that?

File: beets/mediafile.py [hunk 1]
```diff
@@ -1457,6 +1457,7 @@ def delete(self):
         try:
             self.mgfile.delete()
         except NotImplementedError:
+            # (Note: This is fixed in mutagen >=1.31)
```
Comment (jrobeson) [inline beets/mediafile.py]: can you s/Note/FIXME/ here? , that way we have a marker to revisit later.

File: beets/mediafile.py [hunk 2]
```diff
@@ -1960,6 +1960,7 @@ def bitdepth(self):
     def channels(self):
         """The number of channels in the audio (an int)."""
         if isinstance(self.mgfile.info, mutagen.mp3.MPEGInfo):
+            # (Note: MPEGInfo.channels was added in mutagen 1.30)
```
Comment (jrobeson) [inline beets/mediafile.py]: same here,  s/Note/FIXME/

**Comment by sampsyo** [comment]
Fantastic! Thank you, @lazka -- this helps a lot, and I'm thrilled to be able to simplify the exception handling.

I'm tempted to require Mutagen 1.31 and remove those two additional workarounds you noted. If that doesn't raise any red flags for you, I'll make those changes.

**Comment by lazka** [comment]
> I'm tempted to require Mutagen 1.31 and remove those two additional workarounds you noted. If that doesn't raise any red flags for you, I'll make those changes.

Sounds good.

Thanks for the review/merge!

**Pull request merged**